### PR TITLE
Glue Gun

### PR DESCRIPTION
**Pull request by vmiddlemost**
Added glue gun to normal care package loot

### CODE REVIEW

File: JoshMate_TTT2/gamemodes/terrortown/entities/weapons/weapon_jm_zloot_gluegun.lua [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+if CLIENT then
+   SWEP.PrintName          = "Glue Gun"
+   SWEP.Slot               = 6
+
+   SWEP.ViewModelFOV       = 54
+   SWEP.ViewModelFlip      = false
+
+
+   SWEP.Icon               = "vgui/ttt/joshmate/icon_jm_pulsepad.png"
```
Comment (JoshMate) [inline JoshMate_TTT2/gamemodes/terrortown/entities/weapons/weapon_jm_zloot_gluegun.lua]: Needs A unique Icon.

File: JoshMate_TTT2/gamemodes/terrortown/entities/weapons/weapon_jm_zloot_gluegun.lua [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+SWEP.Primary.Delay         = 0.30
+SWEP.Primary.Cone          = 0
+SWEP.Primary.ClipSize      = 5
+SWEP.Primary.DefaultClip   = 5
+SWEP.Primary.ClipMax       = 0
+SWEP.DeploySpeed           = 2
+SWEP.Primary.SoundLevel    = 40
+SWEP.Primary.Automatic     = false
+
+SWEP.Primary.Sound         = "shoot_poisondart.wav"
```
Comment (JoshMate) [inline JoshMate_TTT2/gamemodes/terrortown/entities/weapons/weapon_jm_zloot_gluegun.lua]: Needs A unique sound. (Non Traitors weapons shouldn't have silenced sounds, could cause issues)

File: JoshMate_TTT2/gamemodes/terrortown/entities/weapons/weapon_jm_zloot_gluegun.lua [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+SWEP.Secondary.Sound       = Sound("Default.Zoom")
+SWEP.Kind                  = WEAPON_EQUIP
+SWEP.CanBuy                = {} -- only traitors can buy
+SWEP.WeaponID              = AMMO_GLUEGUN
+SWEP.UseHands              = true
+SWEP.ViewModel             = Model("models/weapons/cstrike/c_rif_famas.mdl")
+SWEP.WorldModel            = Model("models/weapons/w_rif_famas.mdl")
+
+
+local GlueGun_Range         = 1200
```
Comment (JoshMate) [inline JoshMate_TTT2/gamemodes/terrortown/entities/weapons/weapon_jm_zloot_gluegun.lua]: Variables should start with a lowercase, and don't mix cases (variableNameLikeThis or variable_name_like_this) (do as I say not as I do) (:cope)

File: JoshMate_TTT2/gamemodes/terrortown/entities/weapons/weapon_jm_zloot_gluegun.lua [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+   
+   if SERVER then
+      
+      -- JM Changes Extra Hit Marker
+      local hitMarkerOwner = self:GetOwner()
+      JM_Function_GiveHitMarkerToPlayer(hitMarkerOwner, 0, false)
+      -- End Of
+
+      -- Set Status and print Message
+      weaponOwner:ChatPrint("[Glue Gun]: You've glued someone!")
```
Comment (JoshMate) [inline JoshMate_TTT2/gamemodes/terrortown/entities/weapons/weapon_jm_zloot_gluegun.lua]: ChatPrint is deprecated, we now use " JM_Function_PrintChat(weaponOwner, "Equipment", ent:Nick() .. " has been Poisoned!" )"

File: JoshMate_TTT2/gamemodes/terrortown/entities/weapons/weapon_jm_zloot_gluegun.lua [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+      JM_RemoveBuffFromThisPlayer("jm_buff_gluegrenade",ent)
```
Comment (JoshMate) [inline JoshMate_TTT2/gamemodes/terrortown/entities/weapons/weapon_jm_zloot_gluegun.lua:71]: Unless you really want to be directly coupled with the glue grenades balance, I'd suggest creating a new buff that can be balanced independently of the glue grenade. but depends on what you are trying to do.

**Review by JoshMate — CHANGES_REQUESTED** [review]
5 Comments.

File: JoshMate_TTT2/gamemodes/terrortown/entities/weapons/weapon_jm_zloot_gluegun.lua (hunk 1 above)
Comment (vmiddlemost) [reply JoshMate_TTT2/gamemodes/terrortown/entities/weapons/weapon_jm_zloot_gluegun.lua]: Changed icon to a famas with blue backdrop

File: JoshMate_TTT2/gamemodes/terrortown/entities/weapons/weapon_jm_zloot_gluegun.lua [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+   local effect = EffectData()
+   local ePos = ent:GetPos()
+   if ent:IsPlayer() then ePos:Add(Vector(0,0,40))end
+   effect:SetStart(ePos)
+   effect:SetOrigin(ePos)
+   
+   
+   
+   util.Effect("cball_explode", effect, true, true)
```
Comment (JoshMate) [inline JoshMate_TTT2/gamemodes/terrortown/entities/weapons/weapon_jm_zloot_gluegun.lua]: You could replace this "Sparks Effect" with the glue grenade yellow glue splat effect, you may be able to copy and paste the effect code from there, but you may have to tweak arguments a little.

File: JoshMate_TTT2/gamemodes/terrortown/entities/weapons/weapon_jm_zloot_gluegun.lua (hunk 2 above)
Comment (vmiddlemost) [reply JoshMate_TTT2/gamemodes/terrortown/entities/weapons/weapon_jm_zloot_gluegun.lua]: Created and added new "shoot_gluegun.wav" sound to replace place holder sound.

File: JoshMate_TTT2/gamemodes/terrortown/entities/weapons/weapon_jm_zloot_gluegun.lua (hunk 3 above)
Comment (vmiddlemost) [reply JoshMate_TTT2/gamemodes/terrortown/entities/weapons/weapon_jm_zloot_gluegun.lua]: GlueGun_Range -> glueGunRange

File: JoshMate_TTT2/gamemodes/terrortown/entities/weapons/weapon_jm_zloot_gluegun.lua (hunk 4 above)
Comment (vmiddlemost) [reply JoshMate_TTT2/gamemodes/terrortown/entities/weapons/weapon_jm_zloot_gluegun.lua]: Changed line to new line.

File: JoshMate_TTT2/gamemodes/terrortown/entities/weapons/weapon_jm_zloot_gluegun.lua (hunk 5 above)
Comment (vmiddlemost) [reply JoshMate_TTT2/gamemodes/terrortown/entities/weapons/weapon_jm_zloot_gluegun.lua:71]: Leaving this un-changed as I'm happy with keeping the glue gun effect as the same as the glue grenade effect... for now >:)

File: JoshMate_TTT2/gamemodes/terrortown/entities/weapons/weapon_jm_zloot_gluegun.lua (hunk 6 above)
Comment (vmiddlemost) [reply JoshMate_TTT2/gamemodes/terrortown/entities/weapons/weapon_jm_zloot_gluegun.lua]: Changed sparks effect to match glue grenade effect for lore consistency

**Review by JoshMate — APPROVED** [review]

**Pull request merged**